viewing_party/party.py

```diff
@@ -1,23 +1,190 @@
+        return 0.0
+    
+    num_of_ratings = 0
+    ratings_total = 0
+        
+    # for i in range(len(user_data["watched"][0])):
+    #     num_of_ratings += 1
+    #     ratings_total += user_data["watched"][0]["rating"]
+
+    for movie in user_data["watched"]:
+        ratings_total += movie["rating"]
+        num_of_ratings += 1
+    
+    return ratings_total/num_of_ratings
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    
+    genre_frequency = {}
 
+    #first for loop tracks amount of times a genre appears
+    for movie in user_data["watched"]:
+        genre_type = movie["genre"]
+        if genre_type in genre_frequency:
+            genre_frequency[genre_type] += 1
+        else:
+            genre_frequency[genre_type] =1
+    
+    most_watched_genre = None
+    highest_genre_count = 0
+
+    #second for loop picks out which genre is the most watched
+    for genre, count in genre_frequency.items():
+        if count > highest_genre_count:
+            highest_genre_count = count
+            most_watched_genre = genre
+
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_movies_list = []
+    total_friends_watched = []
+
+    #loops through values of key 'friends' in user data, and is appended to total_friends_watched list
+    for movies in user_data["friends"]:
+        total_friends_watched += movies["watched"]
+
+    #loops through values of key 'watched' in user_data to get movies watched
+    for unique_movies in user_data["watched"]:
+        #if statement compares movies in total_friends_watched list to what the user has watched, any unique movie gets appended to unique_movies_list
+        if unique_movies not in total_friends_watched:
+            unique_movies_list.append(unique_movies)
+    
+    return unique_movies_list
+
+
+def get_friends_unique_watched(user_data):
+    unique_movies_list = []
+    total_friends_watched = []
+
+    #loops through values of key 'friends' in user data, and is appended to total_friends_watched list
+    for movies in user_data["friends"]:
+        total_friends_watched += movies["watched"]
+    
+    #loops through movies in the total_friends_watched list to compare (in the if statement) if the movie is not in the user's watched list and hasn't appeared in the unique_movies_list (checks for duplicates) to then append the unique movie
+    for unique_movies in total_friends_watched:
+        if (unique_movies not in user_data["watched"]) and (not unique_movies in unique_movies_list):
+            unique_movies_list.append(unique_movies)
+    
+    return unique_movies_list
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):

```

Comment:
Solid approach! Since we need to consider only movies that one of our friends have watched but we have not, how could we reuse our previous functions `get_friends_unique_watched` to simplify the code?